metpy/calc/basic.py

```diff
@@ -787,6 +787,158 @@ def smooth_n_point(scalar_grid, n=5, passes=1):
     return smooth_grid
 
 
+@exporter.export
+@preprocess_xarray
+def altimeter_to_station_pressure(altimeter_value, height):
+    r"""Convert the altimeter measurement to station pressure.
+
+    This function is useful for working with METARs since they do not provide
+    altimeter values, but not sea-level pressure or station pressure.
+    The following definitions of altimeter setting and station pressure
+    are taken from [Smithsonian1951]_ Altimeter setting is the
+    pressure value to which an aircraft altimeter scale is set so that it will
+    indicate the altitude above mean sea-level of an aircraft on the ground at the
+    location for which the value is determined. It assumes a standard atmosphere.
+    Station pressure is the atmospheric pressure at the designated station elevation.
+    Finding the station pressure can be helpful for calculating sea-level pressure
+    or other parameters.
+
+    Parameters
+    ----------
+    altimeter_value : `pint.Quantity`
+        The altimeter setting value as defined by the METAR or other observation,
+        which can be measured in either inches of mercury (in. Hg) or millibars (mb)
+    height: `pint.Quantity`
+        Elevation of the station measuring pressure.
+
+    Returns
+    --------
+    station_pressure: `pint.Quantity`
+        The station pressure in hPa or in. Hg, which can be used to calculate sea-level
+        pressure
+
+    See Also
+    ---------
+    altimeter_to_sea_level_pressure
+
+    Notes
+    -------
+    This function is implemented using the following equations from the
+    Smithsonian Handbook (1951) p. 269
+
+    Equation 1:
+     .. math:: A_{mb} = (p_{mb} - 0.3)F
+
+    Equation 3:
+     .. math::  F = \left [1 + \left(\frac{p_{0}^n a}{T_{0}} \right)
+                   \frac{H_{b}}{p_{1}^n} \right ] ^ \frac{1}{n}
+
+    Where
+
+    :math:`p_{0}` = standard sea-level pressure = 1013.25 mb
+
+    :math:`p_{1} = p_{mb} - 0.3` when :math:`p_{0} = 1013.25 mb`
+
+    gamma = lapse rate in NACA standard atmosphere below the isothermal layer
+    :math:`6.5^{\circ}C. km.^{-1}`
+
+    :math:`t_{0}` = standard sea-level temperature 288 K
+
+    :math:`H_{b} =` station elevation in meters (elevation for which station
+      pressure is given)
+
+    :math:`n = \frac{a R_{d}}{g} = 0.190284` where :math:`R_{d}` is the gas
+      constant for dry air
+
+    And solving for :math:`p_{mb}` results in the equation below, which is used to
+    calculate station pressure :math:`(p_{mb})`
+
+    .. math:: p_{mb} = \left [A_{mb} ^ n - \left (\frac{p_{0} a H_{b}}{T_0}
+                       \right) \right] ^ \frac{1}{n} + 0.3
+
+    """
+    # Shorten the name for altimeter value
+    altim = altimeter_value
```

Comment:
I'd prefer to just wrap the calculation in the return statement. Just add an extra set of parenthesis. Technically speaking, this assignment will actually require additional bytecode at runtime in the Python interpreter, whereas adding parentheses is free at runtime.